### PR TITLE
Tidy up the repo

### PR DESCRIPTION
**Pull request by HakanVardarr**
### Tidy

- [x] Move `chaincash_*` crates to a `crates/` directory to keep clear separation from other directories
- [x] Use `[workspace.dependencies]` in root `Cargo.toml` to ensure crates use the same version of libraries

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
-  "chaincash_server",
-  "chaincash_cli",
-  "chaincash_store",
-  "chaincash_app",
-  "chaincash_predicate"
+  "crates/chaincash_server",
+  "crates/chaincash_cli",
+  "crates/chaincash_store",
+  "crates/chaincash_app",
+  "crates/chaincash_predicate",
```
Comment (ross-weir) [inline Cargo.toml]:
```suggestion
  "crates/*",
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
 ]
+
+[workspace.dependencies]
```
Comment (ross-weir) [inline Cargo.toml:5]: Could we please only include dependencies that are used by more than one `chaincash` crate as workspace dependencies.

For example, the `diesel` crates will only be used in `chaincash_store` so lets move them back there

**Review by ross-weir — COMMENTED** [review]
Awesome! Thanks so much for the contribution.

Just a couple of small comments :+1:

**Comment by HakanVardarr** [comment]
Is this better?

**Comment by ross-weir** [comment]
LGTM, thanks :rocket:

**Pull request merged**